src/metabase/api/user.clj

```diff
@@ -179,7 +179,45 @@
                (or api/*is-superuser?*
                    api/*is-group-manager?*)
                (hydrate :group_ids))
-     :total  (t2/count User (user-clauses status query group_id include_deactivated))
+     :total  (t2/count User (user-clauses status query (if (some? group_id) [group_id] nil) include_deactivated))
```

Comment:
Do we care that we are using an extra sql call here? I use similar logic in my new API.

Reply:
One extra SQL call is OK, I think. It becomes more of an issue when it's an n+1 situation, where each row in a result set is queried individually, or stuff like that